docstr_coverage/config_file.py

```diff
@@ -23,21 +25,53 @@ def set_config_defaults(ctx, param, value):
     if (value is not None) and os.path.exists(value):
         with open(value) as f:
             config_data = yaml.safe_load(f) or {}
-
-        try:
-            # Check if `paths` were given in config file
-            config_paths = config_data.pop("paths")
-        except KeyError:
-            pass
-        else:
-            # Use config default if `paths` was not provided as CLI argument
-            if not ctx.params.get("paths") and config_paths:
-                if isinstance(config_paths, str):
-                    config_paths = [config_paths]
-
-                # Resolve paths like Click would have with the `click.Path.resolve_path` kwarg
-                ctx.params["paths"] = tuple([os.path.realpath(path) for path in config_paths])
-        finally:
-            ctx.default_map = config_data
+            ctx.params["config_file"] = value
+        # Resolve paths like Click would have with the `click.Path.resolve_path` kwarg
+        _extract_non_default_list(
+            config_data,
+            ctx,
+            "paths",
+            lambda config_paths: tuple([os.path.realpath(path) for path in config_paths]),
+        )
+        _extract_non_default_list(config_data, ctx, "ignore_patterns", lambda x: x)
+        ctx.default_map = config_data
 
     return value
+
+
+def _extract_non_default_list(
+    config_data: Dict, ctx: click.Context, field: str, process: Callable[[List], Any]
+) -> None:
+    """Processes a field of the config file which should not be read as a default value,
```

Comment:
We may have different understandings of this, but it seems like a more appropriate description might be 
```suggestion
Processes a field of the config file which should be used as the default value if not provided as a CLI argument
```

Does that sounds right to you?